tests/my_house_tests.py

```diff
@@ -0,0 +1,128 @@
+    'date': EXPENSE_DATE,
+    'amount': EXPENSE_AMOUNT,
+    'category': EXPENSE_CATEGORY,
+    'description': EXPENSE_BAD_DESCRIPTION,
+}
+EXPENSE_BAD_FORM_DATA_AMOUNT = {
+    'date': EXPENSE_DATE,
+    'amount': EXPENSE_BAD_AMOUNT,
+    'category': EXPENSE_CATEGORY,
+    'description': EXPENSE_DESCRIPTION,
+}
+
+
+@pytest.fixture
+def generate_expense_form():
+    form = ExpenseForm(EXPENSE_FORM_DATA)
+    return form
+
+
+@pytest.fixture
+def generate_bad_description_expense_form():
+    form = ExpenseForm(EXPENSE_BAD_FORM_DATA_DESCRIPTION)
+    return form
+
+
+@pytest.fixture
+def generate_bad_amount_expense_form():
+    form = ExpenseForm(EXPENSE_BAD_FORM_DATA_AMOUNT)
+    return form
+
+
+@pytest.mark.django_db
+class TestMyHouseViews:
+    def test_get_house_view_function(self, client, new_user):
+        client.force_login(new_user)
+        HouseFactory(user=new_user).save()
+        response = client.get('/house/')
+        assert response.status_code == 200
+
+    def test_mine_page_expenses_table_views_templates(self):
+        get_template(MINE_EXPENSES_TABLE_ROUTE)
+
+    def test_mine_page_house_table_views_templates(self):
+        get_template(MINE_HOUSE_TABLE_ROUTE)
+
+    def test_mine_page_mine_page_views_templates(self):
+        get_template(MINE_PAGE_ROUTE)
+
+    def test_mine_sidebar_views_templates(self):
+        get_template(MINE_SIDEBAR_ROUTE)
+
+
+@pytest.mark.django_db
+class TestExpenseForm:
+    def test_expense_form(self, generate_expense_form):
+        form = generate_expense_form
+        assert form.is_valid()
+
+    def test_bad_form_amount(self, generate_bad_amount_expense_form):
+        form = generate_bad_amount_expense_form
+        assert not form.is_valid()
+
+    def test_bad_form_description(self, generate_bad_description_expense_form):
+        form = generate_bad_description_expense_form
+        assert not form.is_valid()
+
+    def test_add_expenses_to_my_house(self, client, new_user):
+        client.force_login(new_user)
+        HouseFactory(user=new_user).save()
+        response = client.post('/house/add_expense/', EXPENSE_FORM_DATA)
+        assert len(Expenses.objects.all()) == 1
+        assert response.status_code == 302
+        assert response.url == '/../house'
+
+    def test_not_add_expenses_to_my_house_with_bad_amount(self, client, new_user):
+        client.force_login(new_user)
+        HouseFactory(user=new_user).save()
+        response = client.post('/house/add_expense/', EXPENSE_BAD_FORM_DATA_DESCRIPTION)
+        assert len(Expenses.objects.all()) == 0
+        assert response.status_code == 200
+
+    def test_not_add_expenses_to_my_house_with_bad_description(self, client, new_user):
+        client.force_login(new_user)
+        HouseFactory(user=new_user).save()
+        response = client.post('/house/add_expense/', EXPENSE_BAD_FORM_DATA_AMOUNT)
+        assert len(Expenses.objects.all()) == 0
+        assert response.status_code == 200
+
```

Comment:
Like with Shelly's The EXPENSE_BAD_FORM_DATA_ are mismatched between the two tests

Reply:
Fixed 👍